xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -133,10 +134,14 @@ ListMLERanker::train(const vector<vector<FeatureVector>>& training_data)
 	throw InvalidArgumentError("Cannot train: no training data");
     int feature_cnt = training_data[0][0].get_fcount();
 
-    // Initialize the parameters for neural network
+    // initialize the parameters for neural network with Xavier initialization.
+    // construct a trivial random generator engine:
+    std::default_random_engine generator(469382313);
+    std::normal_distribution<double> distribution(0.0, sqrt(2.0 /
+							   (1 + feature_cnt)));
```

Comment:
We have `using namespace std;` active in this file, so there's no need for the `std::` qualifiers (which probably means no need to wrap the line above).

`469382313` seems an unexplained magic number - that sort of thing deserves a comment.  I tried a web search for it but found nothing that seemed relevant.

Also we seem to be duplicating this Xavier initialisation code in each ranker - could we not have a single common version of it?

Reply:
fixed in a fixup commit.